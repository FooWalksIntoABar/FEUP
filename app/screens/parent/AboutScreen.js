import React, { Component } from 'react'
import { Body, Header, Icon, Left, Right, Title, Button, Content, Container, Text, H1, H2 } from 'native-base'
import PropTypes from 'prop-types'

export default class AboutScreen extends Component {
  render () {
    return (
      <Container>
        <Header>
          <Left>
            <Button transparent onPress={() => this.props.navigation.pop()}>
              <Icon name='arrow-back' />
            </Button>
          </Left>
          <Body>
            <Title>Sobre a equipa</Title>
          </Body>
          <Right />
        </Header>
        <Content style={{ marginHorizontal: 8, marginVertical: 8 }}>
          <H1>Sobre a equipa</H1>
          <Text style={{ fontSize: 14, paddingBottom: 10 }}>
            Esta aplicação foi feita possível pelo esforço conjunto da equipa de desenvolvimento e fantasticamente pacientes product owners.
          </Text>
          <H2>Product Owners</H2>
          <Text style={{ fontSize: 14, paddingBottom: 10 }}>
            • Carla Oliveira - Mestrado Integrado em Psicologia{'\n'}
            • Mariana Toste - Mestrado Integrado em Psicologia
          </Text>
          <H2>Equipa de Desenvolvimento</H2>
          <Text style={{ fontSize: 14, paddingBottom: 10 }}>
            • Afonso Pinto - @afonsobspinto{'\n'}
            • Francisco Maria - @francismaria{'\n'}
            • João Castro - @joaopscastro{'\n'}
            • Miguel Mano - @aquelemiguel{'\n'}
            • Paulo Correia - @pipas{'\n'}
            • Rui Oliveira - @ruimoliveira{'\n'}
            • Sérgio Salgado - @chaotixkilla
          </Text>
        </Content>
      </Container>
    )
  }
}

AboutScreen.propTypes = {
  navigation: PropTypes.object.isRequired
}
