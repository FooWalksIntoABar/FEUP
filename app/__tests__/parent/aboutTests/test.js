import React from 'react'
import { Button } from 'native-base'
import { configure, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

import AboutScreen from '../../../screens/parent/AboutScreen'

configure({ adapter: new Adapter() })

describe('AboutScreen snapshot', () => {
  it('renders AboutScreen correctly', async () => {
    const wrapper = shallow(<AboutScreen
      navigation={{ navigate: jest.fn(), pop: jest.fn() }} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.find(Button).at(0).props().onPress()
    wrapper.unmount()
  })
})
