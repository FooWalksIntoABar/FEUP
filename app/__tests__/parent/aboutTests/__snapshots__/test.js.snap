// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutScreen snapshot renders AboutScreen correctly 1`] = `
<Styled(Container)>
  <Styled(Header)>
    <Styled(Left)>
      <Styled(Button)
        onPress={[Function]}
        transparent={true}
      >
        <Styled(Icon)
          name="arrow-back"
        />
      </Styled(Button)>
    </Styled(Left)>
    <Styled(Body)>
      <Styled(Title)>
        Sobre a equipa
      </Styled(Title)>
    </Styled(Body)>
    <Styled(Right) />
  </Styled(Header)>
  <Styled(Content)
    style={
      Object {
        "marginHorizontal": 8,
        "marginVertical": 8,
      }
    }
  >
    <Styled(H1)>
      Sobre a equipa
    </Styled(H1)>
    <Styled(Text)
      style={
        Object {
          "fontSize": 14,
          "paddingBottom": 10,
        }
      }
    >
      Esta aplicação foi feita possível pelo esforço conjunto da equipa de desenvolvimento e fantasticamente pacientes product owners.
    </Styled(Text)>
    <Styled(H2)>
      Product Owners
    </Styled(H2)>
    <Styled(Text)
      style={
        Object {
          "fontSize": 14,
          "paddingBottom": 10,
        }
      }
    >
      • Carla Oliveira - Mestrado Integrado em Psicologia
      

      • Mariana Toste - Mestrado Integrado em Psicologia
    </Styled(Text)>
    <Styled(H2)>
      Equipa de Desenvolvimento
    </Styled(H2)>
    <Styled(Text)
      style={
        Object {
          "fontSize": 14,
          "paddingBottom": 10,
        }
      }
    >
      • Afonso Pinto - @afonsobspinto
      

      • Francisco Maria - @francismaria
      

      • João Castro - @joaopscastro
      

      • Miguel Mano - @aquelemiguel
      

      • Paulo Correia - @pipas
      

      • Rui Oliveira - @ruimoliveira
      

      • Sérgio Salgado - @chaotixkilla
    </Styled(Text)>
  </Styled(Content)>
</Styled(Container)>
`;
